### DISCUSSION TITLE
Custom Icons in Speeddial

### DISCUSSION THREAD

**Discussion by Eyowzitgoin** [primefaces/community#1427] [PrimeNG] [answered]
We need to be able to add custom icons into a speeddial component. Not only the open/close buttons, but the buttons contained within the menuitem array. We have a need to maintain an existing icon library which means using Prime Icons is not an option for us. Allowing the templating of either the icon itself or the entire button would give us the freedom we need with this component. Templating the buttons via the menu items would give the added flexibility to those who wish to add text or other things such as text to their button options.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We'll add template options in the future https://github.com/primefaces/primeng/issues/15045
